### PR TITLE
#139 RqHref must be an interface

### PR DESCRIPTION
**Pull request by dmzaytsev**
for #139 

* refactored RqHref as an interface
* created RqHref.Base which implements RqHref

### CODE REVIEW

**Comment by davvd** [comment]
Thanks, let me find someone who can review this pull request

**Comment by davvd** [comment]
@alevohin this pull request is for you, please review

File: src/main/java/org/takes/rq/RqHref.java [hunk 1]
```diff
@@ -24,50 +24,78 @@
...
+    /**
+     * Request decorator, for HTTP URI query parsing.
+     *
+     * <p>The class is immutable and thread-safe.
+     * @author Dmitry Zaytsev (dmitry.zaytsev@gmail.com)
+     * @version $Id$
+     * @since 0.13.1
+     */
+    @EqualsAndHashCode(of = "origin")
+    final class Base implements RqHref {
```
Comment (alevohin) [inline src/main/java/org/takes/rq/RqHref.java]: @dmzaytsev this class should be `public`.

File: src/main/java/org/takes/rq/RqHref.java [hunk 2]
```diff
@@ -24,50 +24,78 @@
...
+        /**
+         * Original request.
+         */
+        private final transient Request origin;
```
Comment (alevohin) [inline src/main/java/org/takes/rq/RqHref.java]: @dmzaytsev why `RqHref.Base` doesn't extend `RqWrap`?

**Comment by alevohin** [comment]
@dmzaytsev thank you! Please see comments above.

File: src/main/java/org/takes/rq/RqHref.java (hunk 1 above)
Comment (dmzaytsev) [inline src/main/java/org/takes/rq/RqHref.java]: @alevohin the modifier public is redundant for inner classes in interfaces, isn't it ?

**Comment by dmzaytsev** [comment]
@alevohin take a look the changes please

File: src/main/java/org/takes/rq/RqHref.java (hunk 1 above)
Comment (alevohin) [inline src/main/java/org/takes/rq/RqHref.java]: @dmzaytsev you are totally right, I agree.

**Comment by alevohin** [comment]
@dmzaytsev looks good, thank you! Lets also change test's javadocs in class `RqHrefTest`. Now it looks like
> RqQuery can ...

It's not true, let's replace it with `RqHref.Base`

**Comment by dmzaytsev** [comment]
@alevohin done

**Comment by alevohin** [comment]
@rultor merge @dmzaytsev thank you!

**Comment by dmzaytsev** [comment]
@alevohin thanks for review!

**Comment by yegor256** [comment]
@rultor merge this

**Comment by davvd** [comment]
@elenavolokhova please, review this ticket for compliance with our [QA rules](http://www.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor deploy

**Comment by elenavolokhova** [comment]
@alevohin One comment regarding addressing the comments:
>davvd commented 7 days ago
Thanks, let me find someone who can review this pull request

Could you please ask everyone to fix them prior to closing the ticket?

**Comment by alevohin** [comment]
@davvd please fix your message
> Thanks, let me find someone who can review this pull request

It should be started with "@dmzaytsev"

**Comment by davvd** [comment]
@alevohin fixed

**Comment by alevohin** [comment]
@davvd thank you!

**Comment by alevohin** [comment]
@elenavolokhova
> @alevohin One comment regarding addressing the comments:
davvd commented 7 days ago
Thanks, let me find someone who can review this pull request
Could you please ask everyone to fix them prior to closing the ticket?

 done

**Comment by elenavolokhova** [comment]
@alevohin Thank you!

**Comment by elenavolokhova** [comment]
@davvd Ticket passed QA review. Quality is acceptable.

**Comment by davvd** [comment]
> @davvd thank you!

@alevohin no problem :)

**Comment by davvd** [comment]
> @davvd Ticket passed QA review. Quality is acceptable.

@elenavolokhova thanks, got it. everybody, please try to make it good next time

**Comment by davvd** [comment]
@alevohin I added 10 mins to @elenavolokhova (for QA review) in transaction 55585960

**Pull request merged**